src/ScottPlot/Settings.cs

```diff
@@ -75,7 +78,12 @@ public PlotDimensions GetPlotDimensions(int xAxisIndex, int yAxisIndex)
             (double yMin, double yMax) = yAxis.Dims.RationalLimits();
             var limits = (xMin, xMax, yMin, yMax);
 
-            return new PlotDimensions(figureSize, dataSize, dataOffset, limits);
+            var dims = new PlotDimensions(figureSize, dataSize, dataOffset, limits);
+            if (OutputWidth.HasValue && OutputHeight.HasValue)
```

Comment:
If we allow the user to pass in values that change the aspect ratio we don't need both of these to be non-null, only one. Might be easiest to let `SetOutputDimensions` take nullable values and have it deal with it.